Sources/Screens/DeveloperMenu/DeveloperMenuViewController.swift

```diff
@@ -40,18 +43,22 @@ final class DeveloperMenuViewController: BaseViewController {
     
     // Makes a successful request to the hello-world endpoint as long as the access token is valid
     private func helloWorldHappyPath() {
-        Task {
-            do {
-                let data = try await networkClient?.makeAuthorizedRequest(exchangeRequest: URLRequest(url: AppEnvironment.stsToken),
-                                                                          scope: "sts-test.hello-world.read",
-                                                                          request: URLRequest(url: AppEnvironment.stsHelloWorld))
-                happyPathResultLabel.showSuccessMessage("Success: \(String(decoding: data!, as: UTF8.self))")
-            } catch let error as ServerError {
-                happyPathResultLabel.showErrorMessage(error)
-            } catch {
-                happyPathResultLabel.showErrorMessage()
+        if userStore.validAuthenticatedUser {
```

Comment:
If the user has opted out of biometrics, this will always fail.  I think we need to be looking at TokenHolder.shared.validAccessToken as well.
I say as well because I just tested using it *instead*, and then it fail if you opt in to biometrics, quit the app, and come back.  As the TokenHolder is a singleton now, I think what should maybe happen is that after tokens are read from the secure store we stash them in the TokenHolder and use the shared instance to reference them in all cases.
Then, if the TokenHolder becomes our source of truth, perhaps it would make sense to store requiresReauth flag there, rather than having it on MainCoordinator.

Reply:
Yep, good shout, forgot about non-local auth users here.
I agree TokenHolder should be our single source of truth for all the tokens.
The problem there is the fact we get our full `TokenResponse` object including all tokens, expiry date etc. from the web flow (and populate the `tokenResponse` property on TokenHolder which `validAccessToken` references).
Though, for a returning user, we only access one token at a time and store them on the top-level properties of TokenHolder, meaning the `tokenResponse` property is nil and `validAccessToken` would be false.
We've talked about this before and how we could/should be retrieving all the tokens from secure store for a returning user but maybe we need to bump the priority of the work to do that in light of this.
As a work around in lieu of that work being done I guess we could put some sort of logic for `userStore.validAuthenticatedUser` OR `TokenHolder.shared.validAccessToken`..?

Reply:
Yes, I think do an OR for now.